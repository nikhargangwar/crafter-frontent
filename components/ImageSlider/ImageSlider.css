@import url('https://fonts.googleapis.com/css2?family=Josefin+Sans&display=swap');

.image-slider {
  width: 60%;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  flex-direction: column;
  background-color: #2b2e30;
  padding: 170px 0;
}

.image-slider h1 {
  font-size: 60px;
  color: white;
  font-family: 'Josefin Sans', sans-serif;
}

.image-slider span {
  font-size: 60px;
  font-weight: bold;
  color: #18ffff;
  font-family: 'Josefin Sans', sans-serif;
}

.image-slider-images {
  height: 40%;
  width: 80%;
  padding: 70px 0;
}

.image-slider-images img {
  height: 400px !important;
}