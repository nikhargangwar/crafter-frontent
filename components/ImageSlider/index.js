import React from 'react';
import './ImageSlider.css';
import ImageGallery from 'react-image-gallery';
import PropTypes from 'prop-types';

function ImageSlider({ text, images, emoji }) {
  return (
    <div className='image-slider'>
      <h1>{text} <span>Developer</span> {emoji}</h1>
      <div className='image-slider-images'>
        <ImageGallery 
          items={images}
          showNav={false}
          showThumbnails={false}
          showFullscreenButton={false}
          showPlayButton={false}
          slideInterval={3000}
          autoPlay={true}
        />
      </div>
    </div>
  );
}

ImageSlider.propTypes = {
  text: PropTypes.string.isRequired,
  images: PropTypes.shape({
    original: PropTypes.string.isRequired
  }),
  emoji: PropTypes.string.isRequired
};

export default ImageSlider;
