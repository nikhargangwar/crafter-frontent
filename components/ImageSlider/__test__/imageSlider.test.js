import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import ImageSlider from '../index.js';
describe('ImageSlider up', () => {
  it('should render correctly', () => {
    const mockProps = { 
      text: 'text', 
      images: ['1', '2'], 
      emoji: 'emoji' 
    };
    const { asFragment } = render(<BrowserRouter><ImageSlider {...mockProps}/></BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});