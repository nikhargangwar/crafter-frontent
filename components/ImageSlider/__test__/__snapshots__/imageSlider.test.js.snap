// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageSlider up should render correctly 1`] = `
<DocumentFragment>
  <div
    class="image-slider"
  >
    <h1>
      text 
      <span>
        Developer
      </span>
       emoji
    </h1>
    <div
      class="image-slider-images"
    >
      <div
        aria-live="polite"
        class="image-gallery"
      >
        <div
          class="image-gallery-content  image-gallery-thumbnails-bottom"
        >
          <div
            class="image-gallery-slide-wrapper  image-gallery-thumbnails-bottom"
          >
            <div
              class="image-gallery-swipe"
            >
              <div
                class="image-gallery-slides"
              >
                <div
                  aria-label="Go to Slide 1"
                  class="image-gallery-slide  image-gallery-center "
                  role="button"
                  style="display: inherit; transform: translate3d(0%, 0, 0);"
                  tabindex="-1"
                >
                  <img
                    alt=""
                    class="image-gallery-image"
                    height=""
                    loading="eager"
                    sizes=""
                    srcset=""
                    title=""
                    width=""
                  />
                </div>
                <div
                  aria-label="Go to Slide 2"
                  class="image-gallery-slide  image-gallery-right "
                  role="button"
                  style="display: inherit; transform: translate3d(100%, 0, 0);"
                  tabindex="-1"
                >
                  <img
                    alt=""
                    class="image-gallery-image"
                    height=""
                    loading="eager"
                    sizes=""
                    srcset=""
                    title=""
                    width=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
